conda_smithy/github.py

```diff
@@ -1,27 +1,24 @@
 import os
 from random import choice
 
+import conda_build.api
+import github
 from git import Repo
-
 from github import Github
 from github.GithubException import GithubException
 from github.Organization import Organization
 from github.Team import Team
-import github
 
-import conda_build.api
 from conda_smithy.utils import get_feedstock_name_from_meta
```

Comment:
We are not being too consistent with relative / absolute imports, I think

Reply:
This is how the new file looks like:

https://github.com/conda-forge/conda-smithy/blob/d8a8d8dab3baa97a61374bc14764995931a30053/conda_smithy/github.py#L1-L14

Seems pretty ordered to me. Note https://github.com/conda-forge/conda-smithy/pull/1919#discussion_r1593447856

Reply:
Sometimes we are doing `from .utils import X` and sometimes `from conda_smithy.utils import X`.

Compare this line to this [other one](https://github.com/conda-forge/conda-smithy/pull/1919/files#diff-27f5c1fcf8e15b2b01490459c0c6589f02d699d19110b6a9a80d66b034f6fc8bR19)

Reply:
Good point. My proposal: Let's ban all relative imports with the `TID` ruleset and `flake8-tidy-imports.ban-relative-imports = "all"`.

I've done this in 4902d5d.

Quote from [PEP 8](https://peps.python.org/pep-0008/#imports):

> Absolute imports are recommended, as they are usually more readable and tend to be better behaved (or at least give better error messages) if the import system is incorrectly configured (such as when a directory inside a package ends up on sys.path):
```python
import mypkg.sibling
from mypkg import sibling
from mypkg.sibling import example
```
> However, explicit relative imports are an acceptable alternative to absolute imports, especially when dealing with complex package layouts where using absolute imports would be unnecessarily verbose:
```python
from . import sibling
from .sibling import example
```

So I think we're in line with PEP 8 since the package layout is not complex at all.